common/eth2/src/lighthouse_vc/http_client.rs

```diff
@@ -263,7 +230,10 @@ impl ValidatorClientHttpClient {
     /// Perform a HTTP GET request, returning `None` on a 404 error.
     async fn get_opt<T: DeserializeOwned, U: IntoUrl>(&self, url: U) -> Result<Option<T>, Error> {
         match self.get_response(url).await {
-            Ok(resp) => self.signed_json(resp).await.map(Option::Some),
+            Ok(resp) => {
+                let body = resp.bytes().await.map_err(Error::from)?;
+                serde_json::from_slice(&body).map_err(Error::InvalidJson)
```

Comment:
I think you need the `.map(Option::Some)` here, otherwise `serde_json` will try to decode an `Option<T>` rather than a `T`. It's probably fine most of the time, but might be weird if the server sent back `null` which could decode to `None`.

Reply:
fixed in f270e02